### DISCUSSION TITLE
MCMC plot modification (font sizes, plot ranges etc)

### DISCUSSION THREAD

**Discussion by gokhannY** [phoebe-project/phoebe2#906] [Questions]
Hey everyone,

I was wondering if it is possible to change some of the properties of MCMC plot, such as font sizes etc. I'm not satisfied how it looks on default. Likewise, I have checked "latex representation" on documents but it's kind of dull for what my needs. 

I use "gnuplot" for making MCMC plots with Phoebe 1.0, since it gives the outcome as a .txt file I can change the fonts etc easily. I would like to do that in Phoebe 2.x too. 

I'm adding an outcome of my last MCMC calculation, you can clearly see it's not pretty. 
[abc.pdf](https://github.com/user-attachments/files/15847200/abc.pdf)

Thanks in advance. 

edit: The results are good. I used wide distribution ranges for some parameters but it doesn't show right away on the graph. For example, inclination, I used "1.0" for so widely it should show like between 87 and 89.

**Comment by kecnry** [comment]
Unfortunately that plot in Phoebe doesn't support much customization through arguments sent to `b.plot(...)`.  But the matplotlib figure (created by the corner package) is returned, so you should be able to modify anything from that object.

**Reply by gokhannY** [reply]
Hey Kyle,

Thanks for advice but I have been trying to manipulate the figure and failed so far. Could you guide me to a website or tutorial, or link to make me how to figure that out?

**Reply by kecnry** [reply]
At that point you will have to look at matplotlib docs or drill into the matplotlib objects.  The returned figure has a `fig.axes` which is a list of the individual subplots.  Call `dir(fig.axes[0])` to see all the available methods/attributes that you can modify.

For example (note that for `b.plot`, two arguments are returned, so you'll need `_, mplfig = b.plot(...)` instead):

```
import phoebe

import matplotlib
matplotlib.use('tkagg')

b = phoebe.default_binary()
b.add_distribution({'teff@primary': phoebe.gaussian_around(100), 'requiv@primary': phoebe.gaussian_around(0.2)})
fig = b.plot_distribution_collection()
print(dir(fig.axes[0]))
fig.axes[0].set_xlim(5000, 7000)
fig.axes[1].set_xlim(5000, 7000)
```

**Reply by gokhannY** [reply]
Thanks a lot Kyle. I was doing something  with `rc.Params` but only managed to change some font sizes etc. this is much more better.
